client/src/components/Admin/Announcements.jsx

```diff
@@ -0,0 +1,405 @@
+                      "& > *": {
+                        borderBottom: "unset",
+                        cursor: "pointer",
+                        backgroundColor: "#efefef",
+                      },
+                    }}
+                    hover
+                  >
+                    <TableCell>
+                      <IconButton
+                        aria-label="edit-announcement"
+                        size="small"
+                        onClick={() => {
+                          setEditAnnouncement(row);
+                          setEditModalOpen(true);
+                        }}
+                      >
+                        <EditIcon />
+                      </IconButton>
+                    </TableCell>
+                    <TableCell align="left" component="th" scope="row">
+                      {row.announcementId}
+                    </TableCell>
+                    <TableCell align="left" component="th" scope="row">
+                      {row.title}
+                    </TableCell>
+                    <TableCell align="left" component="th" scope="row">
+                      {row.description}
+                    </TableCell>
+                    <TableCell>
+                      <FormControlLabel
+                        control={
+                          <Switch
+                            color="success"
+                            checked={row.is_enabled}
+                            onChange={(e) =>
+                              handleIsEnabled(row.announcementId, e.target.checked)
+                            }
+                          />
+                        }
+                        label={
+                          <Typography variant="body2" sx={{ fontSize: "14px" }}>
+                          </Typography>
+                        }
+                      />
+                    </TableCell>
+                    <TableCell align="right">
+                      <IconButton
+                        color="error"
+                        aria-label="delete-announcement"
+                        onClick={async (e) => {
+                          e.stopPropagation();
+                          try {
+                            await announcementService.remove(row.announcementId);
+                            announcementRefetch();
+                          } catch (error) {
+                            console.error(
+                              "Failed to remove announcement:",
+                              error
+                            );
+                          }
+                        }}
+                      >
+                        <DeleteIcon />
+                      </IconButton>
+                    </TableCell>
+                  </TableRow>
+                </React.Fragment>
+              ))}
+          </TableBody>
+        </Table>
+      </TableContainer>
+
+      <Dialog open={announcementModalOpen} onClose={handleModalClose}>
```

Comment:
I think we should add an `aria-labelledby=""` attribute to the dialog components 👀 
you can use this as a reference > https://github.com/hackforla/food-oasis/blob/1b390603c100a6cfd040897e5e580139e5fd2cb8/client/src/components/Admin/AssignDialog.jsx#L25

Reply:
Very good suggestion. I added it to both the dialog components.